sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkExecutor.java

```diff
@@ -299,11 +378,16 @@ public BulkExecutor(CosmosAsyncContainer container,
     }
 
     private Mono<CosmosBulkOperationResponse<TContext>> retryOtherExceptions(
-        CosmosItemOperation itemOperation, Exception exception, FluxSink<CosmosItemOperation> groupSink,
-        CosmosException cosmosException, ItemBulkOperation<?> itemBulkOperation) {
+        CosmosItemOperation itemOperation,
+        Exception exception,
+        FluxSink<CosmosItemOperation> groupSink,
+        CosmosException cosmosException,
+        ItemBulkOperation<?> itemBulkOperation,
+        PartitionScopeThresholds<TContext> thresholds) {
+
         return itemBulkOperation.getRetryPolicy().shouldRetry(cosmosException).flatMap(result -> {
             if (result.shouldRetry) {
-                return this.enqueueForRetry(result.backOffTime, groupSink, itemBulkOperation);
+                return this.enqueueForRetry(result.backOffTime, groupSink, itemBulkOperation, thresholds);
             } else {
                 return Mono.just(BridgeInternal.createCosmosBulkOperationResponse(
```

Comment:
the client 408 probably should add here

Reply:
It does anyway - all invocations of enqueueForRetry will be accounted for.

Reply:
hmm, nope, because it is not readonly operation for 408